### PR TITLE
Add code coverage reporting support

### PR DESCRIPTION
**Pull request by eresende-nuodb**
Add new build target "coverage-smoke" which builds, tests and generates code coverage report.
This code coverage report will be available on CircleCI as artifacts.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -67,6 +101,9 @@ workflows:
       - smoke_tests:
           context:
             - common-config
+      - coverage_smoke:
+          context:
+            - common-config
```
[the remark this answers is not in the document]
Comment (eresende-nuodb) [reply .circleci/config.yml:106]: Good point. I will confirm with Don if they are same and, if so, get rid of smoke-tests.
Thanks.

[the remark this answers is not in the document]
Comment (dwhite-nuodb) [reply .circleci/config.yml:106]: The actual tests are the same, I kept them separate just with the thinking when in a development cycle, I don't need to look at coverage results every time I am trying to test a development change.  Is that not what you usually do?  Do you always run code coverage tools while in the steps of feature development or debugging?

**Review by dwhite-nuodb — APPROVED** [review]
I am good with the changes as is, if you want to convince me we should not have different options for tests without code coverage, then you can try, but it seemed logical to me to have avoid code coverage while intensely trying to get something working.

**Review by adriansuarez — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (adriansuarez) [reply .circleci/config.yml:106]: It is also possible to conditionalize generating coverage stats with a pipeline parameter. But I don't see any reason to not generate coverage stats.

I don't know what mean by "every time I am trying to test a development change". This is automated testing that is triggered by every commit pushed to master. If you want to see how much test coverage new code added with a commit has, you can see it in the report artifact. If you're not interested in the coverage stats, then you can ignore them.

**Comment by eresende-nuodb** [comment]
I will merge as it is for now. Technically speaking, since there's no difference between smoke_tests and coverage_smoke, we could live with just one, but it is not a big deal.

**Pull request merged**